reV/generation/generation.py

```diff
@@ -482,11 +489,85 @@ def time_index(self):
             elif step is not None:
                 time_index = time_index[::step]
 
-            self._time_index = self.handle_leap_ti(time_index,
-                                                   drop_leap=self._drop_leap)
+            time_index = self.handle_lifetime_index(time_index)
+            time_index = self.handle_leap_ti(time_index,
+                                             drop_leap=self._drop_leap)
+
+            self._time_index = time_index
 
         return self._time_index
 
+    def handle_lifetime_index(self, ti):
+        """Adjust the time index if modeling full system lifetime.
+
+        Parameters
+        ----------
+        ti : pandas.DatetimeIndex
+            Time-series datetime index with leap days.
+
+        Returns
+        -------
+        ti : pandas.DatetimeIndex
+            Time-series datetime index.
+        """
+        life_var = "system_use_lifetime_output"
+        lifetime_periods = []
+        for sam_meta in self.sam_metas.values():
+            if life_var in sam_meta and sam_meta[life_var] == 1:
+                lifetime_period = sam_meta["analysis_period"]
+                lifetime_periods.append(lifetime_period)
+            else:
+                lifetime_periods.append(1)
+
+        if any(ltp > 1 for ltp in lifetime_periods):
+            # Collect variables to check that this will work
+            n_unique_periods = len(np.unique(lifetime_periods))
+            array_vars = [
+                var for var, attrs in GEN_ATTRS.items()
+                if attrs['type'] == 'array'
+            ]
+            valid_vars = ['gen_profile', 'cf_profile', 'cf_profile_ac']
+            invalid_vars = set(array_vars) - set(valid_vars)
+            invalid_requests = [var for var in self.output_request
+                                if var in invalid_vars]
+
+            if n_unique_periods != 1:
+                # Only one time index, so lifetime outputs all need to match
+                msg = ('reV cannot handle multiple analysis_periods when '
+                       'modeling with `system_use_lifetime_output` set '
+                       'to 1. Found {} different analysis_periods in the SAM '
+                       'configs'.format(n_unique_periods))
+                logger.error(msg)
+                raise ConfigError(msg)
+
+            elif invalid_requests:
```

Comment:
The `elif` is not necessary here, since code execution terminates when the exception in the above block is raised. Can you please change it to a simple `if` and move it to be just under where `invalid_requests` is defined (should happen automatically once you move the block right above)

Reply:
Done